docs/development/release.md

```diff
@@ -56,15 +56,19 @@ Update the CHANGELOG.md. Put a header under the 'Unreleased' section with the ne
 
 Add a link reference for the github "compare" view at the bottom of the CHANGELOG.md, following the existing pattern. This link reference should create a link in the CHANGELOG's release header to changes in the corresponding release.
 
-### 5. Update Cumulus package API documentation
+### 5. Update DATA_MODEL_CHANGELOG.md
```

Comment:
Added a quick reference to updating the DATA_MODEL_CHANGELOG.  We should really automate this.